packages/xchain-util/src/asset.ts

```diff
@@ -217,7 +226,13 @@ const NON_SYNTH_DELIMITER = '.'
  */
 export const assetFromString = (s: string): Asset | null => {
   const isSynth = s.includes(SYNTH_DELIMITER)
-  const delimiter = isSynth ? SYNTH_DELIMITER : NON_SYNTH_DELIMITER
+  let isTrade = s.includes(TRADE_DELIMITER)
+
+  isSynth ? true : isTrade = false; // Can't be a Synth and Trade at the same time.
```

Comment:
not proper use of the ternary operator. 
you can directly set isTrade to false inside an if statement when isSynth is true
`if (isSynth) {
  // If it's a Synth, it can't be a Trade. Directly enforce this logic.
  isTrade = false;
}`

Reply:
Trunk didn't like it either. 
Updated with an if else statement for clarity.